sphinx_gallery/gen_rst.py

```diff
@@ -1487,3 +1498,141 @@ def save_rst_example(
     # in case it wasn't in our pattern, only replace the file if it's
     # still stale.
     _replace_md5(write_file_new, mode="t")
+
+
+def _get_class(gallery_conf, key):
+    what = f"sphinx_gallery_conf[{repr(key)}]"
+    val = gallery_conf[key]
+    if key == "within_subsection_order":
+        alias = (  # convenience aliases
+            "ExampleTitleSortKey",
+            "FileNameSortKey",
+            "FileSizeSortKey",
+            "NumberOfCodeLinesSortKey",
+        )
+        if val in alias:
+            val = f"sphinx_gallery.sorting.{val}"
+    if isinstance(val, str):
+        if "." not in val:
+            raise ConfigError(
+                f"{what} must be a fully qualified name string, got {val}"
+            )
+        mod, attr = val.rsplit(".", 1)
+        try:
+            val = getattr(importlib.import_module(mod), attr)
+        except Exception:
+            raise ConfigError(
+                f"{what} must be a fully qualified name string, could not import "
+                f"{attr} from {mod}"
+            )
+    if not inspect.isclass(val):
+        raise ConfigError(
+            f"{what} must be 1) a fully qualified name (string) that resolves "
+            f"to a class or 2) or a class itself, got {val.__class__.__name__} "
+            f"({repr(val)})"
+        )
+    return val
+
+
+def _get_callables(gallery_conf, key):
+    # the following should be the case (internal use only)
+    singletons = ("reset_argv", "minigallery_sort_order")
+    assert key in ("image_scrapers", "reset_modules", "jupyterlite") + singletons, key
+    which = gallery_conf[key]
```

Comment:
`gallery_conf` is only used once in this func, here. Could we directly pass the conf value instead? This would avoid needing the
```python
    if key == "jupyterlite":
        which = [which["notebook_modification_function"]]
```

as we could directly pass `gallery_conf['jupyterlite']['notebook_modification_function']`

Reply:
Also for readability what about having a constant outside of the function like

```python
CALLABLE_CONFIGS = {
    'single': ("reset_argv", "minigallery_sort_order", "jupyterlite[notebook_modification_function]"),
    'multi': ("image_scrapers", "reset_modules")
}
```

(as we wouldn't need to use the key to get the from `gallery_config`)

Reply:
Happy to have a crack if you're okay with it but don't have the time

Reply:
It will make all the invocations longer and less DRY. What currently shows up as this:
```
mod_fun, = _get_callables(gallery_conf, "notebook_modification_function")
```
would become
```
mod_fun, = _get_callables([gallery_conf["notebook_modification_function"]], "notebook_modification_function")
```
in the place where it's actually used. Which is less DRY and less readable to me. I'd rather add a bit of complication to the function to keep the in-code invocations cleaner.

> (as we wouldn't need to use the key to get the from gallery_config)

Keep in mind we do need the key inside the function to provide a nicely formatted error message.

> Also for readability what about having a constant outside of the function like

The constant is really only used inside the function so to me it's a step backward to move it outside the function

Reply:
Thats fine. I was thinking it would be nice to be able to update a constant instead of digging inside a function. Much of a muchness though.